livekit-agents/livekit/agents/pipeline/pipeline_agent.py

```diff
@@ -505,16 +507,11 @@ def _on_playout_stopped(interrupted: bool) -> None:
         self._track_published_fut.set_result(None)
 
         while True:
-            await self._speech_q_changed.wait()
-
-            while self._speech_q:
-                speech = self._speech_q[0]
-                self._playing_speech = speech
-                await self._play_speech(speech)
-                self._speech_q.pop(0)  # Remove the element only after playing
-                self._playing_speech = None
-
-            self._speech_q_changed.clear()
+            speech_handle = await self._speech_q.get()
+            self._playing_speech = speech_handle
+            await self._play_speech(speech_handle)
+            self._speech_q.task_done()
+            self._playing_speech = None
```

Comment:
I think the big difference here is that the speech is now removed from the queue before the playout is done